source/merge.d.ts

```diff
@@ -1,5 +1,9 @@
 import {Except} from './except';
 
+type ImproveIntersectionHoverReadabilityShallow<T> = T extends Record<string, unknown> ? { [K in keyof T]: T[K] } : T;
```

Comment:
This looks like https://gist.github.com/ClickerMonkey/a081b990b9b14215141fb6248cef4dc4#file-types-ts-L49-L54 Maybe we should do it like that one and call it `Simplify`? I think it might be useful for other types too, so can you move it to https://github.com/sindresorhus/type-fest/blob/master/source/utilities.d.ts